src/measure_theory/set_integral.lean

```diff
@@ -108,6 +107,14 @@ begin
   ... = ∫ x in s, f x ∂μ : by simp
 end
 
+lemma set_integral_congr_set_ae (hs : measurable_set s) (ht : measurable_set t)
```

Comment:
do you really need the measurability of both sets here? I am pretty sure that measurability of one of them, say `s`, is enough, but I'd even bet that the lemma holds without measurability of any of them.

Reply:
You are right indeed, `restrict_congr_set` was the bit I didn't see. The piecewise related lemmas are not useful anymore, should I delete them ?

Reply:
They can still be useful, so I'd keep them.